### PR TITLE
SSE remove event should trigger a xDS push

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
This is a bug fix, previously when incremental SSE enabled. If only a sse removed, no xDS push happen.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@howardjohn @Nino-K have a look when you have a chance?

File: pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go [hunk 1]
```diff
@@ -149,8 +149,8 @@ func (c *SyntheticServiceEntryController) dispatch(config model.Config, event mo
 }
 
 // RegisterEventHandler registers a handler using the type as a key
+// Note: currently it is not called
```
Comment (Nino-K) [inline pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go]: I think it is actually used [here](https://github.com/istio/istio/blob/fc099f72404973929afebab1e9fec32e9a5e5393/pilot/pkg/serviceregistry/synthetic/serviceentry/discovery.go#L68)

File: pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go [hunk 2]
```diff
@@ -99,17 +100,17 @@ func (c *SyntheticServiceEntryController) List(typ, namespace string) (out []mod
...
+		for _, config := range byNamespace {
+			out = append(out, *config)
+		}
 	}
 
-	byNamespace, ok := c.configStore[namespace]
-	if !ok {
-		return nil, nil
-	}
+	sort.Slice(out, func(i, j int) bool {
```
Comment (Nino-K) [inline pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go]: Just curious why we need to sort by config name?

**Comment by Nino-K** [comment]
lgtm

File: pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go]: To have a determined output, as we do for other configs

File: pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go]: Normally, it is called here
```
		for _, descriptor := range schemas.Istio {
			s.configController.RegisterEventHandler(descriptor.Type, configHandler)
		}
	}
```

However, you removed SSE previously

**Pull request merged**